autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -867,225 +574,636 @@ def refit_single_full(self, X=None, y=None, X_val=None, y_val=None, models=None)
-                for stack_column_prefix in model_loaded.stack_column_prefix_lst:
-                    base_model_name = model_loaded.stack_column_prefix_to_model_map[stack_column_prefix]
-                    self.model_graph.add_edge(base_model_name, model_loaded.name)
+            if self.low_memory:
+                del model
+            models_valid += model_name_trained_lst
 
-        self.save()
-        return copy.deepcopy(self.model_full_dict)
+        return models_valid
 
-    # TODO: Take best performance model with lowest inference
-    def best_single_model(self, stack_name, stack_level):
-        """ Returns name of best single model in this trainer object, at a particular stack_level with particular stack_name.
-
-            Examples:
-                To get get best single (refit_single_full) model:
-                    trainer.best_single_model('refit_single_full', 0)  # TODO: does not work because FULL models have no validation score.
-                To get best single (distilled) model:
-                    trainer.best_single_model('distill', 0)
+    def _train_multi(self, X_train, y_train, X_val, y_val, models: List[AbstractModel], hyperparameter_tune=False, feature_prune=False, stack_name='core', k_fold=None, n_repeats=None, n_repeat_start=0, level=0, time_limit=None) -> List[str]:
         """
-        models = self.models_level[stack_name][stack_level]
-        perfs = [(m, self.model_performance[m]) for m in models if self.model_performance[m] is not None]
-        if not perfs:
-            raise AssertionError('No fit models exist with a validation score to choose the best model.')
-        return max(perfs, key=lambda i: i[1])[0]
-
-    # TODO: Take best performance model with lowest inference
-    def get_model_best(self, can_infer=None, allow_full=True):
-        models = self.get_model_names_all(can_infer=can_infer)
-        if not models:
-            raise AssertionError('Trainer has no fit models that can infer.')
-        perfs = [(m, self.model_performance[m]) for m in models if self.model_performance[m] is not None]
-        if not perfs:
-            model_full_dict_inverse = {full: orig for orig, full in self.model_full_dict.items()}
-            models = [m for m in models if m in model_full_dict_inverse]
-            perfs = [(m, self.model_performance[model_full_dict_inverse[m]]) for m in models if self.model_performance[model_full_dict_inverse[m]] is not None]
-            if not perfs:
-                raise AssertionError('No fit models that can infer exist with a validation score to choose the best model.')
-            elif not allow_full:
-                raise AssertionError('No fit models that can infer exist with a validation score to choose the best model, but refit_full models exist. Set `allow_full=True` to get the best refit_full model.')
-        return max(perfs, key=lambda i: i[1])[0]
+        Train a list of models using the same data.
```

Comment:
could clarify where these models will be accessible in this Trainer

Reply:
added